### PR TITLE
FIX: send email to normalized email owner when hiding emails

### PR DESCRIPTION
**Pull request by SamSaffron**
Previous to this change when both `normalize_emails` and `hide_email_address_taken`
is enabled the expected `account_exists` email was only sent on exact email
matches.

This expands it so it also sends an email to the canonical email owner.

### CODE REVIEW

File: spec/requests/users_controller_spec.rb [hunk 1]
```diff
@@ -834,6 +834,42 @@ def enabled?
...
+                   email: dupe_email,
+                 }
+          end.to_not change { User.count }
+
+          expect(response.status).to eq(200)
+          expect(session["user_created_message"]).to be_present
+
+          # expecting a warning emails
+          expect(Jobs::CriticalUserEmail.jobs.size).to eq(1)
+          expect(Jobs::CriticalUserEmail.jobs.first["args"].first["type"]).to eq("account_exists")
```
Comment (tgxworld) [inline spec/requests/users_controller_spec.rb:869]: We have helpers that can simplify the assertions here which you may want to use.

https://github.com/discourse/discourse/blob/80dcaf1e9842deb5afb4652c8efb57a6aa973c99/spec/support/sidekiq_helpers.rb

Comment (tgxworld) [inline spec/requests/users_controller_spec.rb:869]: Also I think it is also important to assert that the right `user_id` is used to enqueue the Sidekiq job

**Comment by SamSaffron** [comment]
both resolved @tgxworld

**Review by tgxworld — APPROVED** [review]

**Comment by tgxworld** [comment]
@SamSaffron Changes look good :+1:

**Pull request merged**